### PR TITLE
Participants unit test

### PR DESCRIPTION
**Pull request by SithuKyaw-Thitsaworks**
Added unit tests and integration tests for code coverage.

### CODE REVIEW

File: packages/participants-svc/jest.config.js [hunk 1]
```diff
@@ -13,12 +13,12 @@ module.exports = {
     coveragePathIgnorePatterns: ["./src/tmp_files"],
     coverageReporters: ["text", ["json", {file: `../../../coverage/${packageName}-final.json`}]],
     clearMocks: true,
-    coverageThreshold: {
+    /* coverageThreshold: {
```
Comment (elnyry-sam-k) [inline packages/participants-svc/jest.config.js:18]: Can you please add a comment here as a TODO to enable it soon after the initial checks are done of this merge..

File: packages/shared-mocks-lib/src/audit_client_mock.ts [hunk 2]
```diff
@@ -0,0 +1,57 @@
...
+ Names of the original copyright holders (individuals or organizations)
+ should be listed with a '*' in the first column. People who have
+ contributed from an organization can be listed under the organization
+ that actually holds the copyright for their contributions (see the
+ Gates Foundation organization for an example). Those individuals should have
+ their names indented and be marked with a '-'. Email address can be added
+ optionally within square brackets <email>.
+
+ * Coil
+ * - Jason Bruwer <jason.bruwer@coil.com>
```
Comment (elnyry-sam-k) [inline packages/shared-mocks-lib/src/audit_client_mock.ts]: please replace with your's :)

File: packages/participants-svc/jest.config.js (hunk 1 above)
Comment (SithuKyaw-Thitsaworks) [reply packages/participants-svc/jest.config.js:18]: Just added a TODO comment, Sam.

File: packages/shared-mocks-lib/src/audit_client_mock.ts (hunk 2 above)
Comment (SithuKyaw-Thitsaworks) [reply packages/shared-mocks-lib/src/audit_client_mock.ts]: Done, this one too.

File: packages/participants-svc/src/domain/participant_agg.ts [hunk 3]
```diff
@@ -359,7 +360,7 @@
      private _getAuditSecCtx(secCtx: CallSecurityContext): AuditSecurityContext {
          return {
              userId: secCtx.username,
-             role: "", // TODO get role
+             role: "", // Need to fix: get role
```
Comment (elnyry-sam-k) [inline packages/participants-svc/src/domain/participant_agg.ts]: please leave as Todo, unless its addressed completely

File: packages/participants-svc/src/domain/participant_agg.ts [hunk 4]
```diff
@@ -788,30 +775,12 @@
...
-                 `Participant with ID: '${participantId}' is not active.`
-             );
-         }
+
+         const existing = await this._validateParticipantAndRetrieve(participantId);
  
          if (!existing.participantEndpoints) existing.participantEndpoints = [];
  
-         // TODO validate endpoint format
+         // Need Fix: to validate endpoint format
```
Comment (elnyry-sam-k) [inline packages/participants-svc/src/domain/participant_agg.ts]: please leave as Todo, unless its addressed completely

File: packages/participants-svc/src/domain/participant_agg.ts [hunk 5]
```diff
@@ -872,39 +841,20 @@
...
-             !(foundEndpoint = await existing.participantEndpoints.find(
-                 (value: IParticipantEndpoint) => value.id === endpoint.id
-             ))
+             !endpoint.id || !foundEndpoint
          ) {
              throw new EndpointNotFoundError();
          }
  
-         // TODO validate endpoint format
+         // Need Fix: to validate endpoint format
```
Comment (elnyry-sam-k) [inline packages/participants-svc/src/domain/participant_agg.ts]: please leave as Todo, unless its addressed completely

File: test/integration/implementations/participants_repo.test.ts.TMP [hunk 6]
```diff
@@ -0,0 +1,232 @@
...
+ Gates Foundation organization for an example). Those individuals should have
+ their names indented and be marked with a '-'. Email address can be added
+ optionally within square brackets <email>.
+
+ * Gates Foundation
+ - Name Surname <name.surname@gatesfoundation.com>
+
+ * Arg Software
+ - José Antunes <jose.antunes@arg.software>
+ - Rui Rocha <rui.rocha@arg.software>
```
Comment (elnyry-sam-k) [inline test/integration/implementations/participants_repo.test.ts.TMP]: please update

**Review by elnyry-sam-k — APPROVED** [review]

**Pull request merged**